### PR TITLE
Add eslint rules

### PR DESCRIPTION
**Pull request by albertogasparin**

### CODE REVIEW

File: src/eslint/rules/__tests__/exaustive-inject.test.js [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    `,
+    {
+      // it should ignore components specified in options
+      code: `
+        import { Suspense, useState, useMemo } from 'react';
+        import { di } from 'react-magnetic-di';
+        import { Query } from 'react-apollo';
+  
+        function MyComponent() {
+          di(useState);
```
Comment (TimeRaider) [inline src/eslint/rules/__tests__/exaustive-inject.test.js:46]: should this inject `Suspense` as well?

File: src/eslint/rules/__tests__/exaustive-inject.test.js [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+ruleTester.run('exhaustive-inject', rule, {
+  valid: [
+    // it should ignore built-in react components
+    `
+      import { Fragment } from 'react';
+      import { di } from 'react-magnetic-di';
+      import { useQuery } from 'react-apollo';
+
+      function MyComponent() {
+        di(useQuery);
```
Comment (TimeRaider) [inline src/eslint/rules/__tests__/exaustive-inject.test.js:22]: should this inject `Fragment`?

File: src/eslint/rules/__tests__/no-extraneous.test.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    },
+    {
+      // it should inject imported React components
+      code: `
+        import { Suspense, useState } from 'react';
+        import { di } from 'react-magnetic-di';
+        import { Query } from 'react-apollo';
+  
+        function MyComponent() {
+          di(Suspense, useState, Query);
```
Comment (TimeRaider) [inline src/eslint/rules/__tests__/no-extraneous.test.js:57]: this one does inject `Suspense`

File: src/eslint/rules/__tests__/no-extraneous.test.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    ecmaVersion: 6,
+    sourceType: 'module',
+    ecmaFeatures: { jsx: true },
+  },
+});
+
+var ruleTester = new RuleTester();
+ruleTester.run('no-extraneous', rule, {
+  valid: [
+    // it should ignore components/hooks with more than one occurrence
```
Comment (TimeRaider) [inline src/eslint/rules/__tests__/no-extraneous.test.js:15]: i don't understand what that one does :)

File: src/eslint/rules/__tests__/exaustive-inject.test.js (hunk 1 above)
Comment (albertogasparin) [reply src/eslint/rules/__tests__/exaustive-inject.test.js:46]: No as React built-ins are treated specially: by default we `di` only stateful hooks as other things (like `Suspense`, `Fragment`, `useMemo`, ... do not really make sense

File: src/eslint/rules/__tests__/exaustive-inject.test.js (hunk 2 above)
Comment (albertogasparin) [reply src/eslint/rules/__tests__/exaustive-inject.test.js:22]: No, this one is fine, we ignore `Fragment` too

File: src/eslint/rules/__tests__/no-extraneous.test.js (hunk 3 above)
Comment (albertogasparin) [reply src/eslint/rules/__tests__/no-extraneous.test.js]: I've changed this so it does not confuse anymore

File: src/eslint/rules/__tests__/no-extraneous.test.js (hunk 4 above)
Comment (albertogasparin) [reply src/eslint/rules/__tests__/no-extraneous.test.js]: This makes sure you inject only things that are actually used in the component. If you have a long list of `di` and you change some functionality (removing code for instance) you might end up maintaining the imports are the injected dependency will still be marked as used by ESLint (used by `di(...)`).

**Pull request merged**